components/Navigation.tsx

```diff
@@ -15,24 +22,66 @@ export type StackNavigatorType<
   ParamsList extends Record<string, object | undefined>
 > = ReturnType<typeof createStackNavigator<ParamsList>>
 
+/**
+ * Returns an object of functions to navigate to essential app screens.
+ */
+export const useCoreNavigation = () => {
+  // TODO: - Actually Navigate
+  const navigation = useNavigation()
+  return {
+    presentEditEvent: (edit: EditEventFormValues, id?: EventID) => {
+      const formValues = toRouteableEditFormValues(edit)
+      if (id) {
+        navigation.navigate("editEvent", {
+          screen: "editEventForm",
+          params: { ...formValues, id }
+        })
+      } else {
+        navigation.navigate("editEvent", {
+          screen: "createEventForm",
+          params: formValues
+        })
+      }
+    },
+    presentProfile: (
+      id: UserID | UserHandle,
+      method: "navigate" | "replace" = "navigate"
+    ) => {
+      ;(navigation as any)[method]("editEvent", {
+        screen: "userProfile",
+        params: { id, method }
+      })
+    },
+    pushEventDetails: (
+      id: EventID,
+      method: "navigate" | "replace" = "navigate"
+    ) => {
+      // NB: Typescript doesn't like the subscript for some reason, but navigation should always
+      // have a navigate and replace method.
+      ;(navigation as any)[method]("eventDetails", { id, method })
+    },
+    pushAttendeesList: (id: EventID) => {
+      navigation.navigate("eventAttendeesList", { id })
+    }
+  }
+}
+
 /**
  * Base styles that all navigation headers should use.
  */
 export const BASE_HEADER_SCREEN_OPTIONS = {
-  headerStyle: {
-    shadowColor: "transparent",
-    elevation: 0
-  },
   cardStyle: {
     backgroundColor: "white"
   },
+  headerShadowVisible: false,
```

Comment:
So the headers now have a drop shadow? I'm a bit confused at this one

Reply:
This is mainly an iOS problem, where there's a thin outline under the header by default. This prop ensures that shadow is removed.